nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/authorization/AuthorizeControllerServiceReference.java

```diff
@@ -88,9 +91,17 @@ public static void authorizeControllerServiceReferences(final Map<String, String
 
                 // if this descriptor identifies a controller service
                 if (propertyDescriptor.getControllerServiceDefinition() != null) {
-                    final String currentValue = authorizable.getValue(propertyDescriptor);
                     final String proposedValue = entry.getValue();
 
+                    ComponentNode componentNode = (ComponentNode) authorizable.getAuthorizable();
```

Comment:
This is true and was bothering me but I see no good solution to be honest.

I assumed that the authorizable here must be a ComponentNode because we are dealing with its properties. But as far as the code goes, authorizable can be a _wrapper_ for a ComponentNode. And maybe sometimes it can be something unrelated to ComponentNodes. Just not sure if those objects can arrive here.

In fact we only need `PropertyDescriptor getPropertyDescriptor(String name);` in `PropertyConfigurationMapper().mapRawPropertyValuesToPropertyConfiguration`
which is actually available on the `ComponentAuthorizable`. But that interface is not awailable in `PropertyConfigurationMapper`. An idea would be to pull up a `PropertyHolder` interface from the `ComponentAuthorizable` into the `nifi-framework-core-api` but that might be too messy.

Do you have a suggestion?

Reply:
Ah, sorry, I overcomplicated things. There's a trivial solution.
I can simply do
```java
                    String proposedEffectiveValue = new PropertyConfigurationMapper()
                        .mapRawPropertyValuesToPropertyConfiguration(propertyDescriptor, proposedValue)
                        .getEffectiveValue(authorizable.getParameterContext());
```